### PR TITLE
Feature/post interactions

### LINKED ISSUE #8: feature: post component
Create Post component with styling to display on home page (for example)

### LINKED ISSUE #9: feature: reply component
Create reply component

### LINKED ISSUE #21: feature: post interactivity
enables following functionalities:
- Like a post
- comment on post
- copy link to post

### PR DESCRIPTION
**Pull request by claudio-OST**
#### Description
This pull request introduces enhancements to user engagement with posts. It enables users to comment on and like posts, as well as copy post links for easy sharing. These features aim to improve user interaction and extend the reach of posts within the platform.

#### Key Changes
1. **Comments**: Users can now leave comments on posts, fostering community discussion and interaction.
2. **Likes**: Users can show their appreciation for posts through likes, adding a layer of engagement.
3. **Copy Link**: This new feature allows users to easily copy and share post links, enhancing the shareability of content.

#### Technical Details
- Updated `CreateReply` to effectively handle user comments on posts.
- Introduced `UpdateLike` action to manage like states in conjunction with backend services.
- Implemented `CopyLink` component for easy sharing of posts.
- Made UI adjustments and resolved minor bugs to enhance the user experience and interface consistency.

### CODE REVIEW

File: utils/api/request.ts [hunk 1]
```diff
@@ -78,5 +78,11 @@ export const httpRequest = async <T>(
     handleRequestError(res);
   }
 
-  return res.json() as T;
+  const text = await res.text();
```
Comment (malinovic) [inline utils/api/request.ts:81]: Not sure why this change is needed. Any explanation for that?

**Review by malinovic — COMMENTED** [review]
A very very good job Claudio!!!!
Just left a comment for a question that isn't really clear to me. If we can sort that out, I will merge the PR. 👍🏻

File: utils/api/request.ts (hunk 1 above)
Comment (claudio-OST) [reply utils/api/request.ts:81]: Thanks for noticing. This change was implemented to avoid an error in the context of form submit but is no longer relevant (the error does not occur anymore). I have reverted the changes. 

EDIT: This is not correct but has been written in error. The problem occurs with the like function and not with form submit.

Comment (claudio-OST) [reply utils/api/request.ts:81]: Sorry. Just had a quick test again and the like function was broken. I have reverted the reverted changes.

A like action will lead to an error that i do not fully understand but that the mentioned change suppresses.

With the change the request/action returns 

0:["$@1",["development",null]]
1:"$undefined"

and without the change there will be a 500 internal server error.

Comment (malinovic) [reply utils/api/request.ts:81]: Fair enough. I will approve it for now. Can you then create a ticket with this description so we can tackle this issue?

**Review by malinovic — APPROVED** [review]
Great work claudio, approved!
